apps/dashboard/src/components/workflow-editor/steps/base/data-object.tsx

```diff
@@ -0,0 +1,130 @@
+import { useMemo, useState } from 'react';
+import { FieldValues, useFormContext } from 'react-hook-form';
+
+import { Button } from '@/components/primitives/button';
+import { Card, CardContent } from '@/components/primitives/card';
+import { ControlInput } from '@/components/primitives/control-input';
+import { FormField } from '@/components/primitives/form/form';
+import { Input, InputRoot } from '@/components/primitives/input';
+import { useSaveForm } from '@/components/workflow-editor/steps/save-form-context';
+import { useWorkflow } from '@/components/workflow-editor/workflow-provider';
+import { parseStepVariablesToLiquidVariables } from '@/utils/parseStepVariablesToLiquidVariables';
+import { RiAddLine, RiDeleteBin6Line, RiInputField, RiQuestionLine } from 'react-icons/ri';
+import { Link } from 'react-router-dom';
+
+const dataObjectKey = 'data';
+
+const InnerDataObject = ({ field }: { field: FieldValues }) => {
```

Comment:
Can we merge this one to DataObject, i wonder if there is a reason for this inner component?

Reply:
Yes, the inner component uses hooks and it needs to be inside the `render` function because some depend on the field. You cannot use hooks inside a render function so it needs to be a component.